Bio/bgzf.py

```diff
@@ -491,6 +492,20 @@ def _load_bgzf_block(handle, text_mode=False):
         return block_size, data
 
 
+def _is_binary(fileobj):
+    """Determine whether a file was opened in binary mode."""
+    try:
+        fileobj.mode
+    except AttributeError:
+        # io.BytesIO instances do not have a 'mode' in Python 3.10
+        # and there doesn't seem to be an equivalent attribute. The following
+        # is taken from https://stackoverflow.com/questions/44584829/
+        # how-to-determine-if-file-is-opened-in-binary-or-text-mode
+        return isinstance(fileobj, (io.RawIOBase, io.BufferedIOBase))
+    else:
+        return "b" in fileobj.mode.lower()
+
+
```

Comment:
instead you could check if `fileobj.read(0)` returns `b""`. This may be a better test, as ultimately this module relies on `fileobj.read` returning a `bytes` object.

Reply:
I like the reading zero approach, and would not give it a helper function.

Reply:
OK, I changed it to use the comparison with `b""` in both places. I don't know whether this is guaranteed to work on the earliest version of Python supported by BioPython. Certainly in modern 3.x versions `"" != b""`. It might be safest to put the original test (on the `mode` attribute) back in place, and only test `read(0)` in the case of an `AttributeError`. Let me know what you prefer. Thanks for reviewing!